source/extensions/config_subscription/grpc/grpc_stream.h

```diff
@@ -26,15 +26,24 @@ template <class RequestProto, class ResponseProto>
 class GrpcStream : public GrpcStreamInterface<RequestProto, ResponseProto>,
                    public Logger::Loggable<Logger::Id::config> {
 public:
+  // The entry value corresponding to the grpc stream's configuration entry index.
+  enum class ConnectedStateValue {
+    // The first entry in the config corresponds to the primary xDS source.
+    FIRST_ENTRY = 1,
```

Comment:
can they simply be named `primary`, `failover`? :) not trying to block on this, just curious

Reply:
My thinking here is to reduce the abstraction layer penetration of the primary/failover to this class as much as possible (at least from code perspective, but added to the comments for maintainability sake). Generally speaking, a GrpcStream object doesn't need to know whether it is used with primary/failover support or not.
The alternative was to move the value setters to the classes that own the grpc stream, but that seemed to be a higher amount of refactoring (adding callbacks in a few places) that I'm not sure is justified as part of this work.

That said, I'm open to modifying the names if others think it is useful.

Reply:
Thanks for clarification.